lib/client.js

```diff
@@ -158,6 +158,128 @@ class Client {
+   *                      active: true,
+   *                      name: [{ use: 'official', family: ['Coleman'], given: ['Lisa', 'P.'] }],
+   *                      gender: 'female',
+   *                      birthDate: '1948-04-14',
+   *                    }
+   *
+   * // Using promises
+   * fhirClient.create({ resourceType: 'Patient',
+   *                     body: newPatient,
+   *                   })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.create({ resourceType: 'Patient',
+   *                                          body: newPatient,
+   *                                         })
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.body - Information about the patient.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  create(params) {
+    return this.httpClient.post(params.resourceType, params.body);
+  }
+
+  /**
+   * Delete a resource by FHIR id.
+   *
+   * @example
+   *
+   * // Using promises
+   * fhirClient.delete({ resourceType: 'Patient', id: 12345 })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.delete({ resourceType: 'Patient', id: 12345 });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  delete(params) {
+    return this.httpClient.delete(`${params.resourceType}/${params.id}`);
+  }
+
+  /**
+   * Update a resource by FHIR id.
+   *
+   * @example
+   *
+   * const newPatient = { resourceType: 'Patient',
+   *                      active: true,
+   *                      name: [{ use: 'official', family: ['Coleman'], given: ['Lisa', 'P.'] }],
+   *                      gender: 'female',
+   *                      birthDate: '1948-04-14',
+   *                    }
+   *
+   * // Using promises
+   * fhirClient.update({ resourceType: 'Patient', id: 12345, body: newPatient })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.update({ resourceType: 'Patient',
+   *                                          id: 12345,
+   *                                          body: newPatient
+   *                                        });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   * @param {string} params.body - Information about the patient.
```

Comment:
Same as before, `params.resource` and the JSDoc isn't just the patient.

Reply:
Fixing this in a new commit